File: lib/feature/pose/pose_input.dart
import 'package:shus_flutter/common/interface/flutter_image.dart';
import 'package:shus_flutter/common/interface/native_interface_base.dart';

class PoseInput implements NativeSendBase {
  double height;
  FlutterImageBase image;

  PoseInput({required this.height, required this.image});

  @override
  Map<String, dynamic> toData() {
    return {
      "height": height,
      "image": image.toData(),
    };
  }
}

File: android/app/src/main/kotlin/com/example/shus_flutter/feature/pose/PoseInput.kt
package com.example.shus_flutter.feature.pose

import com.example.shus_flutter.common.data.FlutterImage
import java.util.*

class PoseInput {
    var height: Double? = null
    var image: FlutterImage? = null

    constructor(data: HashMap<String, Any>?) {
        if (data == null) return
        height = data["height"] as Double?
        val imageData: HashMap<String, Any>? = data["image"] as? HashMap<String, Any>?
        if (imageData != null) {
            image =  FlutterImage(imageData)
        }
    }
}
